contracts/extensions/MultisigPermissions.sol

```diff
@@ -0,0 +1,551 @@
+    changeVoteFunctionality(_permissionDomainId, _motionId, Vote.Reject, _vote == Vote.Reject);
+
+    if (_vote == Vote.Reject && checkThreshold(_motionId, Vote.Reject)) {
+      cancelFunctionality(_motionId);
+    }
+  }
+
+  function changeVoteFunctionality(
+    uint256 _permissionDomainId,
+    uint256 _motionId,
+    Vote _vote,
+    bool _setTo
+  ) private {
+    if (_setTo) {
+      validateUserPermissions(_permissionDomainId, _motionId);
+    }
+
+    Motion storage motion = motions[_motionId];
+    bytes32 userPermissions = getUserRoles(msgSender(), _permissionDomainId);
+
+    uint8 roleIndex;
+    bool newlyAtThreshold;
+    bool anyBelowThreshold;
+
+    while (uint256(motion.requiredPermissions) >= (1 << roleIndex)) {
+      if ((motion.requiredPermissions & bytes32(1 << roleIndex)) > 0) {
+        // Then the motion requires this permission. Let's check it
+        // If the user is adding a vote but lacks the permission, skip
+        if (uint256(userPermissions) & (1 << roleIndex) == 0 && _setTo) {
+          roleIndex += 1;
+          continue;
+        }
+
+        // Update appropriately if vote changes
+        if (getUserVote(_motionId, msgSender(), roleIndex, _vote) != _setTo) {
+          setUserVote(_motionId, msgSender(), _vote, roleIndex, _setTo);
+
+          if (_setTo) {
+            motionVoteCount[_motionId][_vote][roleIndex] += 1;
+          } else {
+            motionVoteCount[_motionId][_vote][roleIndex] -= 1;
+          }
+
+          if (_vote == Vote.Approve) {
+            emit ApprovalChanged(msgSender(), _motionId, roleIndex, _setTo);
+          } else {
+            emit RejectionChanged(msgSender(), _motionId, roleIndex, _setTo);
+          }
+        }
+
+        if (_vote == Vote.Approve) {
+          uint256 threshold = getDomainSkillRoleThreshold(motion.domainSkillId, roleIndex);
+          if (motionVoteCount[_motionId][_vote][roleIndex] < threshold) {
+            anyBelowThreshold = true;
+          } else if (motionVoteCount[_motionId][_vote][roleIndex] == threshold && _setTo) {
+            newlyAtThreshold = true;
+          }
+        }
+      }
+
+      roleIndex += 1;
+    }
+
+    if (_vote == Vote.Approve) {
+      if (anyBelowThreshold) {
+        delete motion.overallApprovalTimestamp;
+      } else if (newlyAtThreshold) {
+        motion.overallApprovalTimestamp = block.timestamp;
+      }
+    }
+  }
+
+  function cancel(
+    uint256 _motionId
+  ) public motionExists(_motionId) notExecuted(_motionId) notRejected(_motionId) {
+    Motion storage motion = motions[_motionId];
+
+    require(
+      msgSender() == motion.creator ||
+        checkThreshold(_motionId, Vote.Reject) ||
+        block.timestamp > motion.creationTimestamp + 7 days,
+      "colony-multisig-not-enough-rejections"
+    );
+
+    cancelFunctionality(_motionId);
+  }
+
+  function cancelFunctionality(uint256 _motionId) private {
+    Motion storage motion = motions[_motionId];
+    motion.rejected = true;
```

Comment:
This could just be `motions[_motionId].rejected = true` since we don't re-use the variable.